pkg/serializer/split/split.go

```diff
@@ -56,24 +57,24 @@ func init() {
 
 // CheckSizeAndSerialize Check the size of a payload and marshall it (optionally compress it)
 // The dual role makes sense as you will never serialize without checking the size of the payload
-func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, []byte, []byte, error) {
+func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, bool, []byte, []byte, error) {
```

Comment:
I think it would be clearer if this continued to return a single `bool` but it would indicate `mustBeSplit` and would be equal to `mustBeSplit = !(checkSizeCompressed(compressedPayload) && checkSizeUnCompressed(payload))`.

then your checks below are just `if mustBeSplit {}` or `if !mustBeSplit {}`.